docs/api/hardhat/getting-started.md

```diff
@@ -2,242 +2,299 @@
+- Contracts compiled with other compilers will fail to deploy to zkSync Era.
 :::
 
-## Setup
+## Project setup
 
-1. Initialize the project and `cd` into the folder:
+To create a new project run the `create` task of the CLI passing a project name:
 
 ```sh
-mkdir greeter-example
-cd greeter-example
+zksync-cli create demo 
 ```
+This command will create a `demo` folder and clone a Hardhat template project inside it. The downloaded project is already configured and contains all the required plugins.
 
-2. Add the dependencies:
-
-```sh
-yarn init -y
-yarn add -D typescript ts-node @types/node ethers@^5.7.2 zksync-web3 @ethersproject/hash @ethersproject/web hardhat @matterlabs/hardhat-zksync-solc @matterlabs/hardhat-zksync-deploy
-```
-
-The `typescript`, `ts-node` and `@types/node` dependencies are optional - plugins will work fine in a vanilla JavaScript environment. Although, please note that this tutorial _does_ use TypeScript.
-
-::: tip
-If you are using Yarn 2 or later, there may be additional steps to ensure that `TypeScript` works correctly in your editor. For more information, check out [Yarn's official documentation](https://yarnpkg.com/getting-started/editor-sdks).
+::: tip Migrating a project
+If you want to migrate an existing project, please check the [project migration guide](./migrating-to-zksync.md).
 :::
 
-## Configuration
 
-Create the `hardhat.config.ts` file and paste the following code within it:
+## Hardhat configuration
+
+The `hardhat.config.ts` file contains a few lines specific to zkSync Era. The most important are:
 
+Imports the zkSync deployment and compiler plugins.
 ```typescript
 import "@matterlabs/hardhat-zksync-deploy";
 import "@matterlabs/hardhat-zksync-solc";
-
-module.exports = {
-  zksolc: {
-    version: "1.3.8",
-    compilerSource: "binary",
-    settings: {},
-  },
-  defaultNetwork: "zkTestnet",
-  networks: {
-    zkTestnet: {
-      url: "https://testnet.era.zksync.dev", // URL of the zkSync network RPC
-      ethNetwork: "goerli", // Can also be the RPC URL of the Ethereum network (e.g. `https://goerli.infura.io/v3/<API_KEY>`)
-      zksync: true,
-    },
-  },
-  solidity: {
-    version: "0.8.13",
-  },
-};
 ```
 
+Dinamically change the network endpoints of the `zkSyncTestnet` network for local tests. This template project includes a basic unit test in the `/test` folder that runs with the local-setup and can be executed with `yarn test`. 
 
-### How to configure multiple compilation targets
-
-To configure the `hardhat.config.ts` file to target both zkSync Era and other networks, do the following:
+```typescript
+// dynamically changes endpoints for local tests
+const zkSyncTestnet =
+  process.env.NODE_ENV == "test"
+    ? {
+        url: "http://localhost:3050",
+        ethNetwork: "http://localhost:8545",
+        zksync: true
+      }
+    : {
+        url: "https://zksync2-testnet.zksync.dev",
+        ethNetwork: "goerli",
+        zksync: true
+      };
+```
```

Comment:
```suggestion
```

3. The template project also includes a basic unit test in the `/test` folder that runs with the local-setup and can be executed with `yarn test`. 

```

Reply:
I mentioned the tests just because the network details are changed dynamically, but I don't think this should be point 3 of this section "hardhat configuration", cause tests are not really related to hardhat configuration.

Reply:
then we should probably move it all together to the tip below. I'll do the change

Reply:
moved to the tip below